sklearn/utils/estimator_checks.py

```diff
@@ -1915,6 +1917,42 @@ def transform(self, X):
                     shallow_params.items()))
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_set_params(name, estimator_orig):
+    # Check that get_params() returns the same thing
+    # before and after set_params() with some fuzz
+    estimator = clone(estimator_orig)
+
+    params = estimator.get_params()
```

Comment:
I think we should be using deep=False, as we will test each estimator separately ...